### ISSUE TITLE
Fix S3242 FP: Don't raise on public methods in controllers

### ISSUE THREAD

**Issue by NikitaEgorov** [SonarSource/sonar-dotnet#5264]
The rule should not raise on any public method inside any kind of controller (ASP.NET Fx, ASP.NET Core, WebApi, ...)

```csharp
public record ClientContractSearchQuery(string QueryValue) : IRequest<ClientContractSearchDto[]>;
```

![image](https://user-images.githubusercontent.com/10349650/149087100-5ad9276e-580b-4bfe-b579-ac5c31088b5b.png)

**Comment by NikitaEgorov** [comment]
SonarQube 9.2.3.50713

**Comment by pavel-mikula-sonarsource** [comment]
Hi @NikitaEgorov ,

Thank you for reporting this. I confirm it as False Positive.

**Comment by costin-zaharia-sonarsource** [comment]
@NikitaEgorov, @pavel-mikula-sonarsource: could you please elaborate on the **why**? 

I'm not sure I understand why we should handle the public controller methods differently.
